src/TextInput.js

```diff
@@ -1,167 +1,156 @@
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 import constants from './constants';
 
-class TextInput extends Component {
-  constructor(props) {
-    super(props);
-
-    this.id = props.id || idgen();
-
-    if (props.password) {
-      this.id = `password${idgen()}`;
+const TextInput = ({
+  value,
+  children,
+  s,
+  m,
+  l,
+  xl,
+  disabled,
+  noLayout,
+  placeholder,
+  icon,
+  label,
+  inputClassName,
+  success,
+  error,
+  password,
+  email,
+  validate,
+  defaultValue,
+  type,
+  ...props
+}) => {
+  const newId = id => {
+    id = id || idgen();
+
+    if (password) {
+      id = `password${idgen()}`;
     }
 
-    if (props.email) {
-      this.id = `email${idgen()}`;
+    if (email) {
+      id = `email${idgen()}`;
     }
-  }
+    return id;
+  };
 
-  componentDidMount() {
+  const [newValue, setNewValue] = useState(value);
+
+  const inputRef = useRef(null);
+
+  useEffect(() => {
     if (typeof M !== undefined) {
       // eslint-disable-next-line react/prop-types
-      this.props['data-length'] && M.CharacterCounter.init(this.inputRef);
+      props['data-length'] && M.CharacterCounter.init(inputRef);
     }
-  }
+  }, [props]);
 
-  componentDidUpdate(prevProps) {
-    const { value } = this.props;
+  useEffect(() => {
+    M.updateTextFields();
+  }, [value]);
 
-    if (value !== prevProps.value && typeof M !== 'undefined') {
-      M.updateTextFields();
-    }
+  const sizes = { s, m, l, xl };
+
+  let responsiveClasses;
+  if (!noLayout) {
+    responsiveClasses = { col: true };
+    constants.SIZES.forEach(size => {
+      responsiveClasses[size + sizes[size]] = sizes[size];
+    });
   }
 
-  render() {
-    const {
-      children,
-      s,
-      m,
-      l,
-      xl,
-      disabled,
-      noLayout,
-      placeholder,
-      icon,
-      label,
-      inputClassName,
-      success,
-      error,
-      password,
-      email,
-      validate,
-      defaultValue,
-      value,
-      type,
-      ...other
-    } = this.props;
-
-    const sizes = { s, m, l, xl };
-
-    let responsiveClasses;
-    if (!noLayout) {
-      responsiveClasses = { col: true };
-      constants.SIZES.forEach(size => {
-        responsiveClasses[size + sizes[size]] = sizes[size];
-      });
-    }
+  const wrapperClasses = cx('input-field', responsiveClasses);
+
+  let computedType;
+  if (type) {
+    computedType = type;
+  } else if (password) {
+    computedType = 'password';
+  } else if (email) {
+    computedType = 'email';
+  } else {
+    computedType = 'text';
+  }
 
-    const wrapperClasses = cx('input-field', responsiveClasses);
-
-    let computedType;
-    if (type) {
-      computedType = type;
-    } else if (password) {
-      computedType = 'password';
-    } else if (email) {
-      computedType = 'email';
-    } else {
-      computedType = 'text';
-    }
+  const inputProps = {
+    placeholder,
+    type: computedType,
+    id: newId,
```

Comment:
Shouldn't you be called the `newId` function here? Check the snapshot, it says the value of the id is a function and it should be a value, right?